### ISSUE TITLE
Support for inline data providers declaration

### ISSUE THREAD

**Issue by Luxian** [psalm/phpunit-psalm-plugin#38]
I'm getting `UndefinedMethod` and `PossiblyUnusedMethod` false positives when I declare my data providers like this:

```php
/** @dataProvider getTestCases */
```
Because the parser adds an extra space a the end of the method name. 
This can be fixed by changing the [following line](https://github.com/psalm/phpunit-psalm-plugin/blob/69aaf6be9f3edecf0158d08dc073f9867e890387/hooks/TestCaseHandler.php#L146):
```diff
-$apparent_provider_method_name = $class_storage->name . '::' . (string) $provider;
+$apparent_provider_method_name = $class_storage->name . '::' . trim((string) $provider);
```
But this doesn't fix the `PossiblyUnusedMethod` error for that constructor.

Any ideas how to properly fix this?

_PS: I wanted to send a PR with a test, but I can't get `composer test` to run without errors on my machine before doing my changes. Is this a known issue (test failing) that I can ignore or am I doing something wrong?_

**Comment by weirdan** [comment]
There are issues with tests, some of them are fixed in #39 .
